consolidation/consolidate.go

```diff
@@ -115,13 +130,48 @@ func nudge(start, preAggInterval, aggNum uint32) (uint32, int) {
 	postAggInterval := preAggInterval * aggNum
 	var num int
 	var diff uint32
+	// two important principles here:
+	// 1) aggregation buckets have timestamps that are cleanly divisible by postAggInterval.
+	// 2) we never make data pretend to be able to predict the future,
+	//    e.g. when aggregating a spike, the spike should never move to an earlier timestamp
+	//    in other words, data should only ever move to a later timestamp when being aggregated, never to the past.
+	// it follows that the start of (first point going into) an aggregation bucket
+	// is the point that has a timestamp of 1 preAggInterval after the cleanly-divisible-by-postAggInterval timestamp.
+
+	// let's clarify all this with an example.
+	// consider: preAggInterval = 5, aggNum = 4 => postAggInterval = 20
+	// points marked with a * are the first points going into an aggregation interval
+
+	// input   consolidate into    output points
+	// 15      ▼
+	// 20        ►                 20
+	// 25 *    ▼
+	// 30      ▼
+	// 35      ▼
+	// 40        ►                 40
+	// 45 *    ▼
+	// 50      ▼
+	// 55      ▼
+	// 60        ►                 60
+	// 65 *    ▼
+	// 70      ▼
+	// 75      ▼
+	//                             80
+
+	// with our nudging approach, we strip all the points that do not form a full aggregation bucket.
+	// IOW the desired first point must be the the first point of a bucket.
+	// in the example above, it means we strip the first 2 points and start at ts=25 (which will go into ts=40).
+
```

Comment:
I agree that it would make more sense to change Graphite than to change the current MT implementation (Although I wouldn't argue based on the theory of relativity).